python/eos/plot/plotter.py

```diff
@@ -377,77 +377,65 @@ class Observable(BasePlot):
         def __init__(self, plotter, item):
             super().__init__(plotter, item)
 
-        def plot(self):
-            item = self.item
-            oname = item['observable']
+            from .specs import Specs, ParametersSpec, KinematicsSpec, RangeSpec, ObservableSpec, VariableSpec
+            self.specs = Specs([
+                ObservableSpec(name='observable'),
+                VariableSpec(name='variable'),
+                RangeSpec(name='range'),
+                KinematicsSpec(name='kinematics', optional=True, default={}),
+                ParametersSpec(name='parameters', optional=True, default={}),
+            ])
+            # TODO: parent class should 'consume' used parameters, such that they are not passed
+            # to the child
 
-            obs_entry = eos.Observables._get_obs_entry(oname)
-            valid_kin_vars = [kv for kv in obs_entry.kinematic_variables()]
-            eos.info('   plotting EOS observable "{}"'.format(oname))
+        def plot(self):
+            item = self.item.copy()
 
-            # create kinematics
             kinematics = eos.Kinematics()
-            if 'kinematics' in item:
-                for k, v in item['kinematics'].items():
-                    if k not in valid_kin_vars:
-                        raise ValueError("Kinematic quantity '" + k + "' does not " +
-                        "match known ones for observable '" + oname + "': " + valid_kin_vars.__repr__())
-                    kinematics.declare(k, v)
-
-            # create parameters
             parameters = eos.Parameters.Defaults()
-            if 'parameters' in item and 'parameters-from-file' in item:
-                eos.warn('    overriding values read from \'parameters-from-file\' with explicit values in \'parameters\'')
-
-            if 'parameters-from-file' in item and type(item['parameters-from-file']) is str:
-                eos.warn('    overriding parameters from file')
-                parameters.override_from_file(item['parameters-from-file'])
-
-            if 'parameters' in item and type(item['parameters']) is dict:
-                for key, value in item['parameters'].items():
-                    parameters.set(key, value)
-
-            # create (empty) options
             options = eos.Options()
 
+            # process user input
+            # need oname for some specs
+            try:
+                oname = item['observable']
+            except KeyError:
+                raise ValueError("Missing key: 'observable'")
+            self.specs['variable'].observable_name = oname
```

Comment:
This is not very elegant, but explicit and safe; forgetting these steps results in an implementation error that will be caught in the notebooks.

Reply:
Why do you rely on the ``KeyError`` exception?

```python3
if not 'observable' in item:
    raise ValueError(...)
```